Lib/test/test_except_star.py

```diff
@@ -0,0 +1,270 @@
+
+import sys
+import unittest
+
+class TestInvalidExceptStar(unittest.TestCase):
+    def test_mixed_except_and_except_star_is_syntax_error(self):
+        errors = [
+            "try: pass\nexcept ValueError: pass\nexcept* TypeError: pass\n",
+            "try: pass\nexcept* ValueError: pass\nexcept TypeError: pass\n",
+            "try: pass\nexcept ValueError as e: pass\nexcept* TypeError: pass\n",
+            "try: pass\nexcept* ValueError as e: pass\nexcept TypeError: pass\n",
+            "try: pass\nexcept ValueError: pass\nexcept* TypeError as e: pass\n",
+            "try: pass\nexcept* ValueError: pass\nexcept TypeError as e: pass\n",
+            "try: pass\nexcept ValueError: pass\nexcept*: pass\n",
+            "try: pass\nexcept* ValueError: pass\nexcept: pass\n",
+        ]
+
+        for err in errors:
+            with self.assertRaises(SyntaxError):
+                compile(err, "<string>", "exec")
+
+    def test_except_star_ExceptionGroup_is_runtime_error_single(self):
+        with self.assertRaises(TypeError):
+            try:
+                raise OSError("blah")
+            except *ExceptionGroup as e:
```

Comment:
That's a good question, we started with ``except *E``, but I found myself gravitating towards ``except* E`` because it feels like the ``*`` is operating on the except and not on the E.   I find ``except*`` pretty, but then I wonder if this will be confusing:


```
>>> try:
...    raise ValueError
... except (*TypeError, *ValueError):
...    pass
...
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
TypeError: Value after * must be an iterable, not type
>>>
```

Well - "except *ValueError" doesn't care that ``*`` is operating on a non-iterable.

Maybe ``except*`` is really a new token (should we think again about catch?).

Reply:
my 2c: I like `except *E` better because it resonates with argument unpacking syntax. I think that symmetry is what makes this proposal attractive in the first place.

Reply:
Also, this is subtle and subjective, but `except *` doesn't look like a whole new syntax to me -- it's merely about modifying the existing construct to support unpacking. Whereas `except*` does feel like a new syntax, which perhaps will raise the bar of accepting this PEP a tad.

Reply:
I have two concerns with where we are at the moment. One is what I said above about naming `except*`- is the unpacking analogy valid? The syntax makes it look like we're unpacking the type or type tuple.

The second issue is about semantics: We add `except*` because we can't change `except`, but during the sprint we had the idea of adding `catch` with a view to it possibly being a replacement for except. Let's imagine we do this and rename `except*` to `catch`, and then users who write new code can just use catch and forget about except.   Then we need to not wrap plain exceptions by an exception group, so 

```
try:
   raise ValueError(12)
catch ValueError as e:
   """e is ValueError(12), not ExceptionGroup("", ValueError(12))"""
```

Con: the runtime type of e is less certain. Pro: You can use catch for all your exception handling needs (so you don't need to learn about except if you're writing new code).

Reply:
I'm not so keen on `catch`, I think it will cause confusion, and as you say the type becomes uncertain. But I think it deserves a place in the rejected ideas section.

Reply:
IMO changing the syntax to `catch` will make the PEP dead in the water. It instantly raises questions like "why do we have two different keywords" etc. I'm very -1 on `catch`.

> One is what I said above about naming except*- is the unpacking analogy valid? The syntax makes it look like we're unpacking the type or type tuple.

That's fair, I did stretch the analogy there a bit. Still, unpacking-like syntax interpretation resonates with me, personally, better.